temporal-sdk/src/main/java/io/temporal/client/WorkflowStartOperationUpdate.java

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.client;
+
+import io.temporal.internal.common.JavaLambdaUtils;
+import io.temporal.workflow.Functions;
+import io.temporal.workflow.UpdateMethod;
+import java.lang.invoke.MethodHandleInfo;
+import java.lang.invoke.SerializedLambda;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.concurrent.*;
+
+public class WorkflowStartOperationUpdate<R>
+    implements StartWorkflowOperation, Future<UpdateHandle<R>> {
```

Comment:
Implements `Future` directly to avoid an extra method call.

Reply:
Yeah the other methods on the `Future` interface will cause confusion and restrict our design in the future